multi_index_map_derive/src/lib.rs

```diff
@@ -27,727 +29,58 @@ pub fn multi_index_map(input: proc_macro::TokenStream) -> proc_macro::TokenStrea
-        let index_name = format_ident!("_{}_index", f.ident.as_ref().unwrap());
-        let ty = &f.ty;
-
-        let (ordering, uniqueness) = get_index_kind(f).unwrap_or_else(|| {
-            abort_call_site!("Attributes must be in the style #[multi_index(hashed_unique)]")
-        });
-
-        match uniqueness {
-            Uniqueness::Unique => match ordering {
-                Ordering::Hashed => quote! {
-                    #index_name: ::multi_index_map::rustc_hash::FxHashMap<#ty, usize>,
-                },
-                Ordering::Ordered => quote! {
-                    #index_name: ::std::collections::BTreeMap<#ty, usize>,
-                },
-            },
-            Uniqueness::NonUnique => match ordering {
-                Ordering::Hashed => quote! {
-                    #index_name: ::multi_index_map::rustc_hash::FxHashMap<#ty, ::std::collections::BTreeSet<usize>>,
-                },
-                Ordering::Ordered => quote! {
-                    #index_name: ::std::collections::BTreeMap<#ty, ::std::collections::BTreeSet<usize>>,
-                },
-            },
-        }
-    });
-
-    // For each indexed field generate a TokenStream representing initializing the lookup table.
-    // Used in `with_capacity` initialization
-    // If lookup table data structures support `with_capacity`, change `default()` and `new()` calls to
-    //   `with_capacity(n)`
-    let lookup_table_fields_init: Vec<::proc_macro2::TokenStream> = fields_to_index()
-        .map(|f| {
-            let index_name = format_ident!("_{}_index", f.ident.as_ref().unwrap());
-            let (ordering, _uniqueness) = get_index_kind(f).unwrap_or_else(|| {
-                abort_call_site!("Attributes must be in the style #[multi_index(hashed_unique)]")
-            });
-            match ordering {
-                Ordering::Hashed => quote! {
-                    #index_name: ::multi_index_map::rustc_hash::FxHashMap::default(),
-                },
-                Ordering::Ordered => quote! {
-                    #index_name: ::std::collections::BTreeMap::new(),
-                },
-            }
-        })
-        .collect();
-
-    // For each indexed field generate a TokenStream representing reserving capacity in the lookup table.
-    // Used in `reserve`
-    // Currently `BTreeMap::extend_reserve()` is nightly-only and uses the trait default implementation,
-    //   which does nothing.
-    // Once this is implemented and stabilized, we will use it here to reserve capacity.
-    let lookup_table_fields_reserve: Vec<::proc_macro2::TokenStream> = fields_to_index()
-        .map(|f| {
-            let index_name = format_ident!("_{}_index", f.ident.as_ref().unwrap());
-            let (ordering, _uniqueness) = get_index_kind(f).unwrap_or_else(|| {
-                abort_call_site!("Attributes must be in the style #[multi_index(hashed_unique)]")
-            });
+    let fields_to_index = named_fields
```

Comment:
Additionally in this PR we now generate all the fields to index at the start and reuse this list, rather than regenerating each time.